lib/components/note-preview/index.tsx

```diff
@@ -29,12 +30,41 @@ type Props = OwnProps & StateProps & DispatchProps;
 export const NotePreview: FunctionComponent<Props> = ({
   editNote,
   fontSize,
+  isFocused,
   note,
   noteId,
   searchQuery,
 }) => {
   const previewNode = useRef<HTMLDivElement>();
 
+  useEffect(() => {
+    const copyRenderedNote = (event: ClipboardEvent) => {
+      if (!isFocused) {
+        return true;
+      }
+
+      // Only copy the rendered content if nothing is selected
+      if (!document.getSelection().isCollapsed) {
+        return true;
+      }
+
+      // const node = document.createDocumentFragment();
+      const div = document.createElement('div');
+      renderToNode(div, note!.content, searchQuery);
+      // node.appendChild(div);
+
+      // TODO this is a bug
+      console.log(div); // innerHTML shows up here in console
+      console.log(div.innerHTML); // innerHTML is now an empty string???????
```

Comment:
looks like `renderToNode()` runs inside a Promise, meaning that `div.innerHTML` shouldn't be populated at least until this function returns and the Promise `.then()` executes, setting the node's `innerHTML`. this is why it works in the console - because the Promise `.then()` has already run (it probably takes microseconds after this outer function finishes).

we can hook into the `Promise` chain by modifying `renderToNode()` to return the `Promise` instead of just running it (and returning `undefined`)

```diff
  export const renderToNode = (
      …
- ) => {
-   renderNoteToHtml(content)
+ ) => renderNoteToHtml(content)
  …
-   });
+ });
- }
```